docs/reference/runtime/Execution.md

```diff
@@ -1,5 +1,9 @@
 # Execution
 
+Embedded systems are dedicated to performing a single duty. They are typically constrained in resources and contain only the necessary components to perform that duty. Most embedded systems are powered by a single microcontroller chip, which is typically limited in resources such as memory. Software for an embedded system must be carefully designed to make the best use of these limited resources. Mbed OS in its default configuration is often too large for constrained systems, but supports a bare metal mode that might be a better option for your target system.
```

Comment:
I don't particularly like the first sentence. Embedded systems can be very complex.

I would rewrite the last sentence to mention the `full profile` versus the `bare metal profile` and emphasize that the latter is targeted at ultraconstrained devices.